tools/ci/tc/decision.py

```diff
@@ -101,14 +100,22 @@ def get_extra_jobs(event):
     return jobs
 
 
-def filter_jobs(jobs, event):
-
-    # Exclude browser stability tests for exports from that specific browser.
+def filter_excluded_users(tasks, event):
+    # Some users' pull requests are excluded from tasks,
+    # such as pull requests from automated exports.
     try:
-        if event["pull_request"]["user"]["login"] == "chromium-wpt-export-bot":
-            jobs.discard("wpt-chrome-dev-stability")
-        if event["pull_request"]["user"]["login"] == "moz-wptsync-bot":
-            jobs.discard("wpt-firefox-nightly-stability")
+        submitter = event["pull_request"]["user"]["login"]
+        excluded_tasks = []
+        for name, task in list(tasks.items()):
+            for excluded_user in task.get("exclude-users", []):
+                if excluded_user == submitter:
+                    excluded_tasks.append(name)
+                    tasks.pop(name)
+        if excluded_tasks:
+            logger.info(
+                f"Tasks excluded for user {submitter}:\n * " +
+                "\n * ".join(excluded_tasks)
+            )
     except KeyError:
```

Comment:
Can you limit the try block to just the first line? If some of the other lines raise a `KeyError` (can't see how now, but maybe later) that shouldn't be suppressed.

Reply:
Yes, that is a better implementation - updated